lib/number.js

```diff
@@ -0,0 +1,319 @@
+import {
+	getDocumentLocaleSettings,
+	getLanguage,
+	merge,
+	validateFormatValue
+} from './common.js';
+
+function formatPositiveInteger(value, descriptor) {
+	value = Math.floor(value);
+
+	const valueStr = '' + value;
+	let ret = '';
+
+	const groupSizes = [3];
+	let currentGroupSizeIndex = -1;
+	const maxGroupSizeIndex = groupSizes.length - 1;
+	let currentGroupSize = 0;
+	let groupEnd = valueStr.length;
+
+	while (groupEnd > 0) {
+		if (currentGroupSizeIndex < maxGroupSizeIndex) {
+			currentGroupSize = groupSizes[++currentGroupSizeIndex];
+		}
+
+		let chunk = null;
+		if (currentGroupSize === 0) {
+			chunk = valueStr.substring(0, groupEnd);
+		} else {
+			const groupStart = groupEnd - currentGroupSize;
+			chunk = valueStr.substring(groupStart, groupEnd);
+		}
+
+		// not first or only chunk
+		if (groupEnd !== valueStr.length) {
+			ret = descriptor.symbols.group + ret;
+		}
+
+		ret = chunk + ret;
+
+		groupEnd -= chunk.length;
+	}
+
+	return ret;
+}
+
+function formatExponentialDecimal(value) {
+	// Get a value should be like "1.25e-8"
+	const pieces = value.split('e-');
+	let ret = pieces[0].replace('.', '');
+	const zeroCount = parseInt(pieces[1]) - 1;
+
+	for (let i = 0; i < zeroCount; i++) {
+		ret = '0' + ret;
+	}
+	return ret;
+}
+
+function validateFormatOptions(options) {
+
+	options = options || {};
+
+	if (options.style !== 'decimal' && options.style !== 'percent') {
+		options.style = 'decimal';
+	}
+	options.minimumFractionDigits = validateInteger(
+		'minimumFractionDigits',
+		options.minimumFractionDigits,
+		0,
+		0,
+		20
+	);
+	options.maximumFractionDigits = validateInteger(
+		'maximumFractionDigits',
+		options.maximumFractionDigits,
+		Math.max(options.minimumFractionDigits, 3),
+		0,
+		20
+	);
+
+	if (options.minimumFractionDigits > options.maximumFractionDigits) {
+		throw new RangeError('maximumFractionDigits value is out of range.');
+	}
+
+	return options;
+
+}
+
+function validateInteger(name, value, defaultValue, min, max) {
+
+	if (value === undefined || value === null) {
+		value = defaultValue;
+	}
+	if (typeof value === 'string') {
+		value = parseInt(value);
+	}
+	if (isNaN(value) || typeof value !== 'number' || (min !== undefined && value < min) || (max !== undefined && value > max)) {
+		throw new RangeError(name + ' value is out of range.');
+	}
+
+	return value;
+
+}
+
+export function getNumberDescriptor() {
+
+	const language = getLanguage();
+
+	const subtags = language.split('-');
+	const baseLanguage = subtags[0];
+
+	let negativePattern = '-{number}';
+	if (baseLanguage === 'ar') {
+		negativePattern = '{number}-';
+	}
+
+	let percentPattern = '{number} %';
+	let percentNegativePattern = '-{number} %';
+	switch (baseLanguage) {
+		case 'es':
```

Comment:
Nit: Another case where `es` and `es-MX` is different

Reply:
Fixed the `es`/`es-MX` case. The Swedish one I'm not uncertain about as the [polyfill](https://github.com/andyearnshaw/Intl.js/blob/master/locale-data/json/sv-SE.json#L282) doesn't reverse the percent symbol at all. I'm going to check with Jeff S... maybe we messed up our LMS data.

Reply:
I think there may also be discrepancies between the decimal and group symbols for those two?

Reply:
Fixed. I'm going to add these base locale tests back into the v2 branch just to make sure I didn't miss anything else.

Reply:
Sounds good, I am definitely not confident in my quick spot-checking